### PR TITLE
Added a unit test for CommunityController as of issue #101. Needed to…

### LINKED ISSUE #101: CommunityController unit tests
## Task description

There is a need to create unit tests for `CommunityController`.
As an example, you can have a look at `UserControllerTest`.

## Task Relationships
This task is a sub-task of: #100

### PR DESCRIPTION
**Pull request by edwardUL99**
… add Data annotation to many response classes but otherwise got 100% coverage on CommunityController

## 🚀 Description

Only changes to the main code was to add @Data annotations to some response classes as junit assertEquals would fail otherwise as it would simply check if the objects are equal by reference, not value

Added many unit tests to achieve 100% coverage on CommunityController

Closes #101 

## 📄 Motivation and Context
Issue #101 requested this. Unit tests were required for CommunityController

## 🧪 How Has This Been Tested?
Ran the unit tests and checked coverage. Ran ./gradlew build to ensure everything built correctly also

## 📦 Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [X] My code follows the code style of this project(Do your best to follow code styles. If none apply just skip this).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/com/myhome/controllers/response/AddCommunityAdminResponse.java [hunk 1]
```diff
@@ -27,6 +28,7 @@
 @AllArgsConstructor
 @Getter
 @Setter
+@Data
```
Comment (DDmit04) [inline src/main/java/com/myhome/controllers/response/AddCommunityAdminResponse.java:31]: `@Data` includes `@Getter`, `@Setter`, `@RequiredArgsConstructor`, `@ToString`, `@EqualsAndHashCode` so you can remove  `@Getter` and `@Setter` annotations

**Comment by DDmit04** [comment]
will test pass if replace `@Data` with `@EqualsAndHashCode`?

**Comment by edwardUL99** [comment]
I didn't realise that. Thanks for the heads up👍

**Comment by edwardUL99** [comment]
> will test pass if replace `@Data` with `@EqualsAndHashCode`?

Yes they pass, so I have replaced `@Data` with `@EqualsAndHashCode`

**Comment by DDmit04** [comment]
grate

**Comment by mslowiak** [comment]
> > will test pass if replace `@Data` with `@EqualsAndHashCode`?
> 
> Yes they pass, so I have replaced `@Data` with `@EqualsAndHashCode`

@edwardUL99 , @DDmit04 Why we are not using just @Data annotation in every class there? Are there any problems with that?

File: src/main/java/com/myhome/controllers/response/ListMemberPaymentsResponse.java [hunk 2]
```diff
@@ -27,14 +27,13 @@
...
 @NoArgsConstructor
-@Getter
-@Setter
 @Data
 public class ListMemberPaymentsResponse {
   private Set<MemberPayment> payments = new HashSet<>();
 
   @Getter
   @Setter
+  @Data
```
Comment (jmprathab) [inline src/main/java/com/myhome/controllers/response/ListMemberPaymentsResponse.java]: `@Data` in itself includes `@Getter` and `@Setter`

**Comment by DDmit04** [comment]
@mslowiak @jmprathab If we using `@Data`, may be should replace all `@Setter` and `@Getter` in other classes by it? Otherwise, using a different set of annotations can be confusing

**Comment by jmprathab** [comment]
> @mslowiak @jmprathab If we using `@Data`, may be should replace all `@Setter` and `@Getter` in other classes by it? Otherwise, using a different set of annotations can be confusing
 
+1 for this, but this needs to be done carefully. `@Data` will introduce `@EqualsAndHashCode` annotation and we might run into cyclic dependency problem and we will have to properly add `exclude="field_name"` to the required entities.

**Comment by mslowiak** [comment]
> > @mslowiak @jmprathab If we using `@Data`, may be should replace all `@Setter` and `@Getter` in other classes by it? Otherwise, using a different set of annotations can be confusing
> 
> +1 for this, but this needs to be done carefully. `@Data` will introduce `@EqualsAndHashCode` annotation and we might run into cyclic dependency problem and we will have to properly add `exclude="field_name"` to the required entities.

Let's do this :)

**Comment by edwardUL99** [comment]
> > > @mslowiak @jmprathab If we using `@Data`, may be should replace all `@Setter` and `@Getter` in other classes by it? Otherwise, using a different set of annotations can be confusing
> > 
> > 
> > +1 for this, but this needs to be done carefully. `@Data` will introduce `@EqualsAndHashCode` annotation and we might run into cyclic dependency problem and we will have to properly add `exclude="field_name"` to the required entities.
> 
> Let's do this :)

Could you explain the Cyclic Dependency Problem? At the moment I just added the Data annotation to the response classes nothing else

**Comment by jmprathab** [comment]
> Could you explain the Cyclic Dependency Problem? At the moment I just added the Data annotation to the response classes nothing else

For instance, when modelling Bi-Directional relationship between `Community` and `CommunityHouse`, we will have `Community` object in `CommunityHouse` and `CommunityHouse` object in `Community` models. In this case, `@EqualsAndHashCode` implementation of Lombok will recursively call hashCode method on the object and we will run into StackOverflow exception.

Ref: https://github.com/rzwitserloot/lombok/issues/1007#issuecomment-175030540

**Comment by edwardUL99** [comment]
> > Could you explain the Cyclic Dependency Problem? At the moment I just added the Data annotation to the response classes nothing else
> 
> For instance, when modelling Bi-Directional relationship between `Community` and `CommunityHouse`, we will have `Community` object in `CommunityHouse` and `CommunityHouse` object in `Community` models. In this case, `@EqualsAndHashCode` implementation of Lombok will recursively call hashCode method on the object and we will run into StackOverflow exception.
> 
> Ref: [rzwitserloot/lombok#1007 (comment)](https://github.com/rzwitserloot/lombok/issues/1007#issuecomment-175030540)

Oh ok, learn something new every day! :)

**Comment by edwardUL99** [comment]
> > > Could you explain the Cyclic Dependency Problem? At the moment I just added the Data annotation to the response classes nothing else
> > 
> > 
> > For instance, when modelling Bi-Directional relationship between `Community` and `CommunityHouse`, we will have `Community` object in `CommunityHouse` and `CommunityHouse` object in `Community` models. In this case, `@EqualsAndHashCode` implementation of Lombok will recursively call hashCode method on the object and we will run into StackOverflow exception.
> > Ref: [rzwitserloot/lombok#1007 (comment)](https://github.com/rzwitserloot/lombok/issues/1007#issuecomment-175030540)
> 
> Oh ok, learn something new every day! :)

I created a separate branch to replace getters and setters with the @Data and setting exclude where appropriate and all tests are passing. However, I am not sure if this is a false sense of security that everything is ok, since if you are excluding a field (for example HouseMember's field CommunityHouse then a Housemember with all the same attributes bar the community house will be shown to be equal even though they clearly are not. Or is it guaranteed to always have unique memberIds to make this a non-issue?

**Comment by jmprathab** [comment]
> I created a separate branch to replace getters and setters with the @DaTa and setting exclude where appropriate and all tests are passing. However, I am not sure if this is a false sense of security that everything is ok, since if you are excluding a field (for example HouseMember's field CommunityHouse then a Housemember with all the same attributes bar the community house will be shown to be equal even though they clearly are not. Or is it guaranteed to always have unique memberIds to make this a non-issue?

This is not a problem. As of now, we are generating `memberId` as a UUID. So, unless there is a UUID collision we are not going to have a problem.

Your changes in the other branch should be good to go.

**Comment by edwardUL99** [comment]
@jmprathab Can I merge this?

**Comment by jmprathab** [comment]
I was waiting for @mslowiak to approve. It looks good for me.

**Pull request merged**